src/app/users/update-contact-alias.ts

```diff
@@ -0,0 +1,31 @@
+import { UsersRepository } from "@services/mongoose"
+import { ValidationError } from "@domain/errors"
+
+export const updateContactAlias = async ({
+  userId,
+  username,
+  alias,
+}: {
+  userId: UserId
+  username: string
+  alias: string
+}): Promise<User | ApplicationError> => {
+  const repo = UsersRepository()
+  const user = await repo.findById(userId)
+  if (user instanceof Error) {
+    return user
+  }
+
+  const found = user.contacts.find((contact) => contact.username === username)
+  if (!found) {
+    return new ValidationError(`User doesn't have contact ${username}`)
+  }
+  found.alias = alias as ContactAlias
```

Comment:
We mentioned it, but no further decision was made. This casting here is probably ok to work with for now though, since that's what we're doing in most other places.

Reply:
On looking at this again though, I know we had said sometime before that we should ideally be checking user inputs before passing them forward. Would `alias` be a candidate for this here?

As far as I can tell no checks were done before it gets to this `persist` point (unless we are doing things in the GraphQL layer somewhere).